### DISCUSSION TITLE
robots.txt, CMS gets indexed by Google

### DISCUSSION THREAD

**Discussion by sensedrive** [directus/directus#8556] [Archive]
Hello everyone,

is it planed to have a robots.txt for Directus, or are there any recomendations to prevent Google (and any other search engines) from indexing the CMS?

Thank you  in advance!

**Comment by rijkvanzanten** [comment]
@sensedrive Down to have a robots.txt for the app! Should be possible to add it to https://github.com/directus/directus/tree/main/app/public and have it work automagically.

Wanna open a PR for that? 🙂

**Reply by calebfaruki** [reply]
Unfortunately, this file doesn't seem to do anything. Tested using local docker container on directus v9.12.1

<img width="1128" alt="Screen Shot 2022-06-04 at 4 54 53 PM" src="https://user-images.githubusercontent.com/5914427/172009311-02f0d392-b0ee-4878-8426-920e53c500b2.png">

**Comment by pavelstianko** [comment]
What is the best practice for robots.txt in CMS? When I have default Directus robots file, it will disallow loading all content (images, api calls). Some of my frontend pages are using clientside fetch and by blocking this (by robots.txt) the Google doesnt load any content to page or in some cases it will return error page. Should I use only `Disallow: /admin/`?

**Reply by br41nslug** [reply]
You should probably also disallow any API endpoints like `/items/*`, `/users/*` and such or better yet keep disallow everything and specifically allow what you want indexed
```
Disallow: /
Allow: /assets/
```

**Reply by pavelstianko** [reply]
Thank you. I need /items/* for client-side fetch, so I will Allow it as well with /assets/

**Reply by rijkvanzanten** [reply]
> I need /items/* for client-side fetch

Curious: Why do you need them in robots.txt for client-side fetch? Isn't that what CORS controls rather than robots.txt?

**Reply by pavelstianko** [reply]
Hey Rijk, all works well in browser, but if I check page using Google Search console I get error, that fetch to directus api is blocked by robots.txt and page preview is only error page (because of blocked request)

**Reply by rijkvanzanten** [reply]
Ah gotcha 👍🏻

**Reply by pavelstianko** [reply]
I think that Google can't make request because this URL is blocked in Directus robots.txt, so I need to change default robots or change all requests to server-side.
